Client and event input validation

cowlinn opened the pull request:
Could use some input on this:
Currently the maximum length of someone's name is 750 characters (I googled)
and the maximum `description` name for an `event` is limited to 1200 characters.

Both numbers are quite arbitrary (and high), should I lower them?

marcusczh commented:
If you want to, you could possibly limit the length of names to something more reasonable like 100 characters.

As for the description for events, that would depend on the intended usage of descriptions, but for display purposes, maybe something like 200 characters could be suitable so that it does not take up too much space on the GUI.

Some testing might be required to see how it would look on the GUI

cowlinn commented:
![image](https://user-images.githubusercontent.com/88648894/198177161-3d9f73b4-1a56-4ae5-b623-404b968aa7ee.png)

Good point! I've added more input validation for phone number as well (and changed the display to show information row by row for person, and event, policy and assignedPolicy (@Ssatu you might wanna check whether everything's in order but it's just to `toString()` ). 

Phone number range [3,25]
Name range [1, 100]
Desc range (for events) [1, 175]

Ssatu commented:
> If you want to, you could possibly limit the length of names to something more reasonable like 100 characters.
> 
> As for the description for events, that would depend on the intended usage of descriptions, but for display purposes, maybe something like 200 characters could be suitable so that it does not take up too much space on the GUI.
> 
> Some testing might be required to see how it would look on the GUI

Agree with this!

Ssatu commented on src/main/java/longtimenosee/model/event/Description.java:
```diff
@@ -9,13 +9,13 @@
     public static final String MESSAGE_CONSTRAINTS = "Description should only contain "
-        + "alphanumeric characters and spaces.";
+        + "alphanumeric characters and spaces. It should not be empty";
+    public static final String LENGTH_CONSTRAINTS = "Ensure description is of an appropriate length";
```
Maybe it would be clearer to the user if we attach the MAXIMUM_DESCRIPTION_LENGTH to the LENGTH_CONSTRAINTS message. Something like  "Ensure description is of an appropriate length of: " MAXIMUM_DESCRITION LENGTH + " characters."

Ssatu reviewed (commented):
LGTM! Just some minor changes for clarity of error messages to users.
